Desktop/components/JASP/Widgets/FileMenu/Database.qml

```diff
@@ -0,0 +1,379 @@
+
+                    x:				dbHostnameLabel.width
+                    width:			dbHostnameInput.width
+                }
+            }
+
+            Item
+            {
+                anchors.left:		parent.left;
+                anchors.right:		parent.right;
+                height:				dbHostnameItem.height
+
+                Text
+                {
+                    id:						dbPasswordLabel
+                    text:					qsTr("Password")
+                    width:					dbHostnameLabel.width
+                    anchors.verticalCenter: parent.verticalCenter
+                }
+
+                PrefsTextInput
+                {
+                    id:						dbPasswordInput
+                    nextEl:					connectButton
+                    text:					fileMenuModel.database.password
+                    onTextChanged:			fileMenuModel.database.password = text;
+
+                    textInput.echoMode:		TextInput.Password
+
+                    x:						dbHostnameLabel.width
+                    width:					dbHostnameInput.width
+                }
+            }
+
+            RoundedButton
+            {
+                id:					connectButton
+                text:				qsTr("Connect to database")
+                onClicked:			fileMenuModel.database.connect();
+                KeyNavigation.tab:	dbQuery.textInput
+            }
+
+
+            RowLayout
+            {
+                width:						parent.width
+
+                enabled:					fileMenuModel.database.connected
+
+                Text
+                {
+                    id:						dbQueryLabel
+                    text:					qsTr("Query")
+                    width:					implicitWidth + jaspTheme.generalAnchorMargin
+                    //anchors.verticalCenter: parent.verticalCenter
+                }
+
+                PrefsTextInput
+                {
+                    id:						dbQuery
+                    nextEl:					runQuery
+                    text:					fileMenuModel.database.query
+                    onTextChanged:			fileMenuModel.database.query = text
+                    LQ.Layout.fillWidth:	true
+                }
+
+                RoundedButton
+                {
+                    id:						runQuery
+                    text:					qsTr("Execute")
+                    onClicked:				fileMenuModel.database.runQuery();
+                    KeyNavigation.tab:		loadResults
+                }
+            }
+
+        }
+
+        PrefsGroupRect
+        {
+            id:			previewGroup
+            title:		qsTr("Preview data")
+            enabled:	fileMenuModel.database.connected && fileMenuModel.database.resultsOK
+
+            Item
+            {
+                height:			childrenRect.height
+                anchors
+                {
+                    left:		parent.left
+                    right:		parent.right
+                    margins:	jaspTheme.generalAnchorMargin
+                }
+
+                RoundedButton
+                {
```

Comment:
If I change the query, execute it, and load it again, then it does not change the data in JASP.